src/modules/SuSEFirewall4Network.rb

```diff
@@ -218,28 +235,24 @@ def AnyInterfacesHandledByFirewall
     # @return	[Boolean] if successful
     def ProtectByFirewall(interface, zone, protect_status)
       # Adding protection
-      if protect_status == true
-        Builtins.y2milestone(
-          "Enabling firewall because of '%1' interface",
-          interface
-        )
-        SuSEFirewall.AddInterfaceIntoZone(interface, zone)
-        SuSEFirewall.SetEnableService(true)
-        SuSEFirewall.SetStartService(true)
-        # Removing protection
+      if protect_status
+        log.info("Enabling firewall because of '#{interface}' interface")
+
+        SuSEFirewall.AddInterfaceIntoZone(interface, zone) if !iface_in_zone?(interface, zone)
+
+        start_fw_service
```

Comment:
no bcs functions ```SetStartService``` and ```SetEnableService``` checks if the service was started / enabled already.

Problematic functions are ```AddInterfaceIntoZone```, ```RemoveInterfaceFromZone``` and most probably others too which updates fw setup unconditionally. Most probably the check would be placed in these functions. However it changes behavior of these functions (can but need not to be a problem). It would also make implementation of this feature a bit more complex. And last but not least an additional layer of check never hurts in network ;-)